src/main/java/ru/skypro/hw/hwCollections3/service/EmployeeBookServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package ru.skypro.hw.hwCollections3.service;
+
+import org.springframework.stereotype.Service;
+import ru.skypro.hw.hwCollections3.domein.Employee;
+import ru.skypro.hw.hwCollections3.exception.DepartmentNotFoundException;
+import ru.skypro.hw.hwCollections3.exception.EmployeeNotFoundException;
+import ru.skypro.hw.hwCollections3.exception.EmployeeStorageIsFullException;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+public class EmployeeBookServiceImpl implements EmployeeBookService {
+    private final int MAX_EMPLOYEE = 6;
+
+    private Map<String, Employee> employees = new HashMap<>(Map.of(
+
+            "Ivan1 Ivanov", new Employee("Ivan1", "Ivanov", 50_000, 1),
+            "Ivan2 Ivanov", new Employee("Ivan2", "Ivanov", 45_000, 1),
+            "Ivan3 Ivanov", new Employee("Ivan3", "Ivanov", 35_000, 2),
+            "Ivan4 Ivanov", new Employee("Ivan4", "Ivanov", 60_000, 2)
+    ));
+
+    @Override
+    public Employee add(String firstName, String lastName, Integer salary, Integer dept) {
+        String key = firstName + " " + lastName;
+        if (MAX_EMPLOYEE < employees.size()) {
+            throw new EmployeeStorageIsFullException();
+        } else if (employees.containsKey(key)) {
+            throw new EmployeeNotFoundException();
+        }
+        Employee employee = new Employee(firstName, lastName, salary, dept);
+        employees.put(key, employee);
+        return employee;
+    }
+
+    @Override
+    public Employee remove(String firstName, String lastName) {
+        String key = firstName + " " + lastName;
+        if (!employees.containsKey(key)) {
+            throw new EmployeeNotFoundException();
+        }
+        Employee employee = employees.get(key);
+        employees.remove(key);
+        return employee;
```

Comment:
не нужно делать get, remove уже возвращает удаленное значение
```suggestion
        return employees.remove(key);
```